### PR TITLE
VACMS-18464 Show all topics by default on Resources & Support landing page

### PR DESCRIPTION
**Pull request by randimays**
## Summary
The new Family Member and Caregiver Benefit Hub was recently published and we want users to be able to see the links on the Resources & Support landing page (under "Browse by topic").

Currently, that section is truncated down to 6 link lists, so we are removing the truncation and the "Show all topics" button completely from the Lists of Links component.

It was determined that the Resources & Support landing page is the only prod page where this component is used so we can safely modify it for that route specifically (`/resources`). 

Ticket comments for context:
- https://github.com/department-of-veterans-affairs/va.gov-cms/issues/18464#issuecomment-2272132968
- https://github.com/department-of-veterans-affairs/va.gov-cms/issues/18464#issuecomment-2276409136

## Related issue(s)
- https://github.com/department-of-veterans-affairs/va.gov-cms/issues/18464

## Testing done
**This can be tested on Tugboat**: https://web-w4azz2khpzdi53uymp1gxkpi0fbqdlqu.demo.cms.va.gov/resources/

1. Check `/resources` locally to be sure there are link lists matching the selected categories in Drupal for this page:

Screenshot 2024-08-05 at 1 34 31 PM

2. Be sure sort order (alphabetically) of categories is preserved; note that links within each category are not alphabetized and should match the order shown in prod
3. Click some (or all) of the links within the sections to be sure link behavior is preserved
4. Validate that there is no "Show all topics" button on `/resources`
5. Check mobile for the above steps

## Screenshots

<details><summary>DESKTOP - Resources & Support landing page</summary>
Screenshot 2024-08-05 at 2 10 38 PM
</details>

<details><summary>MOBILE - Resources & Support landing page</summary>
Screenshot 2024-08-05 at 1 47 54 PM
Screenshot 2024-08-05 at 1 48 07 PM
Screenshot 2024-08-05 at 2 11 00 PM
</details>

## What areas of the site does it impact?

Resources & Support landing page (`/resources`)

## Acceptance criteria

- [x] New topic has capability of being shown without hidden content shown after button is clicked
- [x] Sort order is preserved
- [x] Design review
- [ ] Accessibility review

### CODE REVIEW

File: src/site/paragraphs/lists_of_links.drupal.liquid [hunk 1]
```diff
@@ -1,94 +1,78 @@
 <!-- Section header -->
 <h2 data-template="paragraphs/lists_of_links">{{ entity.fieldSectionHeader }}</h2>
-
-<div class="vads-u-display--flex vads-u-flex-direction--column medium-screen:vads-u-flex-direction--row medium-screen:vads-u-flex-wrap--wrap" data-template="paragraphs/lists_of_links">
   {% assign formattedVaParagraphs = entity.fieldVaParagraphs | formatVaParagraphs %}
+  
+  <!-- R&S landing page should include all categories, none truncated -->
```
Comment (randimays) [inline src/site/paragraphs/lists_of_links.drupal.liquid:5]: This change encapsulated the code for `/resources` inside the parent div (the one with the `data-template` on it). All of the truncation and "Show all topics" logic begins on line 18.

**Comment by randimays** [comment]
@laflannery @thejordanwood This is ready for your review please. Tugboat here for testing since RIs are down: https://web-w4azz2khpzdi53uymp1gxkpi0fbqdlqu.demo.cms.va.gov/resources/

**Review by chriskim2311 — APPROVED** [review]
LGTM!

**Comment by randimays** [comment]
Whoohoo, review instance is up!

**Review by thejordanwood — APPROVED** [review]
Looks good!

**Comment by laflannery** [comment]
@randimays the RI isn't working anymore for me so I used the Tugboat, 2 things:

1. The order of the categories isn't matching the FE but I'm not sure if maybe this was reordered since it was built. 

**Drupal**
image
 **Front end** 
image


2. I didn't know the "Go to all articles" was managed in Drupal. These are duplicative links and it would be better if they were unique - For example, "Go to all disability articles", Go to all burials and memorials articles", etc. Are we able to change that within the scope of this ticket or would this need approve from folks?

**Comment by randimays** [comment]
@laflannery Thanks for the review! 

1. If I understood the ticket correctly, we are to maintain the existing sort order (alphabetical) on the front end.

> Drupal [editing screen](https://prod.cms.va.gov/node/8296/edit) shows listed in user-specified order. FE is alphabetical with exception of first and last ones, and truncated to limit to 6 categories.

I think Fran's comment [here](https://github.com/department-of-veterans-affairs/va.gov-cms/issues/18464#issuecomment-2256574255) (and Daniel's subsequent reply) confirm this.

> 2. I didn't know the "Go to all articles" was managed in Drupal. These are duplicative links and it would be better if they were unique - For example, "Go to all disability articles", Go to all burials and memorials articles", etc. Are we able to change that within the scope of this ticket or would this need approve from folks?

I have no problem with it being part of the scope of this ticket and wouldn't think there'd be any pushback but I think we should confirm with @FranECross for awareness.

@laflannery Are you ok if we merge this PR either way? Those Drupal changes will start at prod and propagate down through lower environments so we won't necessarily benefit from holding this PR until they do.

**Comment by laflannery** [comment]
Gotcha, yes that all makes sense. I shall approve, thanks!

**Review by laflannery — APPROVED** [review]

**Pull request merged**